File: backend-dotnet/SahayaSetu/Models/ResourceContribution.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SahayaSetu.Models
{
    public class ResourceContribution
    {
        [Key]
        public int ContributionId { get; set; }

        [Required]
        public int DonorId { get; set; }

        [ForeignKey(nameof(DonorId))]
        public Donor Donor { get; set; } = null!;

        [Required]
        public int ResourceRequestId { get; set; }

        [ForeignKey(nameof(ResourceRequestId))]
        public ResourceRequest ResourceRequest { get; set; } = null!;

        [Required]
        public int Quantity { get; set; }

        public DateTime ContributedAt { get; set; } = DateTime.UtcNow;
    }
}

File: backend-java/sahayaSetu/src/main/java/com/sahayaSetu/entities/ResourceContribution.java
package com.sahayaSetu.entities;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "resource_contributions")
@Data
public class ResourceContribution {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "contribution_id")
    private Long contributionId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "donor_id", nullable = false)
    private Donor donor;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "resource_request_id", nullable = false)
    private ResourceRequest resourceRequest;

    @Column(nullable = false)
    private Integer quantity;

    @CreationTimestamp
    @Column(name = "contributed_at", updatable = false)
    private LocalDateTime contributedAt;
}